docs/features/interceptors.md

```diff
@@ -61,6 +61,35 @@ namespace System.Runtime.CompilerServices
 }
 ```
 
+#### PROTOTYPE(ic): Use an assembly attribute instead?
+The main reason we want something like `[Interceptable]` is so that users can better reason about *which calls might be getting intercepted or not*. We don't think there's a meaningful *security* concern which is addressed by `[Interceptable]`--if something untrusted has the ability to add source files to your build, taking away the ability for them to intercept certain calls doesn't really make a difference.
+
+It's been suggested that `[Interceptable]` on method declarations is the wrong point of control. Some SG authors will have good reason to want to intercept a call to a method from a library they don't own--for example, EF wanting to intercept calls to IQueryable extensions. It shouldn't be necessary for those authors to have to request the library authors to add an attribute before they can start doing it.
+
+One option to address this is to remove `[Interceptable]` from the design and not provide any mechanism for limiting which calls can be intercepted or not.
+
+Another option would be to expose an assembly attribute instead.
+
+```cs
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>If present, indicates the set of types whose methods can have their calls intercepted in the current compilation.</summary>
+    /// <remarks>When this attribute is not present, methods on any type can have their calls intercepted.</remarks>
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
+    public sealed class AllowInterception(params Type[] types) : Attribute { }
+}
+
+[assembly: AllowInterception(typeof(EndpointRouteBuilderExtensions))]
+```
+
+In order to intercept a classic extension method call, the extension type must be provided to this attribute, not the `this` parameter type.
+
+We specify `AllowMultiple = false` here to imply that the attribute is supposed to be hand-authored by the user. It's not meant for generators to use this to *opt themselves in* to being able to intercept things--it's for the user to state once and comprehensively which things can be intercepted.
+
+In such a scheme, it's not clear how often users would actually want to go to the trouble of writing the attribute--particularly since the process of writing and maintaining it would simply be to run source generators, review the "not interceptable" errors, and add the related types to this attribute. Is that useful to users in practice?
```

Comment:
I do not think it is useful.

We can add this attribute in future if we find actually need for it. It would not be breaking change to add it in future.

Reply:
To be clear, are you in favor of the below option?

> One option to address this is to remove `[Interceptable]` from the design and not provide any mechanism for limiting which calls can be intercepted or not.

Reply:
Yep

Reply:
I agree it is an item we could add later if there was need.

Reply:
I haven't heard any objection to eliminating `InterceptableAttribute` for the .NET 8 version of the feature, and possibly introducing some limiting mechanism in the future.

I'll merge this PR as-is, but have added a note to the test plan to go ahead and eliminate `InterceptableAttribute` in the feature branch in a subsequent PR.

Reply:
Yup, let's get rid of it. Thanks.